TaxCalculator/TaxCalculator.cs

```diff
@@ -47,29 +47,108 @@ public void GetFinancialInformationFromUser()
             IdentifyExemptIncome();
         }
 
+        public void CalculateTaxEstimation()
+        {
+            double untaxedIncome = (_grossIncome + _spouseGrossIncome) - _taxExemptIncome;
+            long totalIncome = _grossIncome + _spouseGrossIncome;
+            double taxEstimate = 0;
+
+            // Select the correct tax bracket based on marriage status
+            var taxBrackets = _isMarried ? _married2019TaxBrackets : _single2019TaxBrackets;
+
+            long previousBracketIncomeLimit = 0;
+            foreach (var bracket in taxBrackets)
+            {
+                long bracketRange = bracket.IncomeLimit - previousBracketIncomeLimit;
+
+                double applicableUntaxedIncome;
+
+                // How much income are we taxing in this bracket?

```

Comment:
The first two comments are not very clear. It is really hard for me to know what is going by just looking at the comments. Maybe come up with a different way of saying what you are trying to say. It seems to me that those comments sound a lot like what YOU personally were thinking at the moment; almost like following your thought process. I suggest you come up with comments that describe the type of calculation you are executing in that particular part. Keep it action-oriented.

Reply:
Ok, good feedback.

Give me suggestions of what you would put for the comments...